sys/posix/sockets/posix_sockets.c

```diff
@@ -221,6 +221,7 @@ static int _sockaddr_to_ep(const struct sockaddr *address, socklen_t address_len
                 return -1;
             }
             struct sockaddr_in *in_addr = (struct sockaddr_in *)address;
+            memset(out, 0, *out);
```

Comment:
I would rather leave this to the caller of this (internal) function, i.e., the calling function should init the struct as needed.

Reply:
thereby also avoiding the separate memset for ipv4 and ipv6 here.

Reply:
But multiplying the memset for every call of this function (which is for).

```
sys/posix/sockets/posix_sockets.c:static int _sockaddr_to_ep(const struct sockaddr *address, socklen_t address_len,
sys/posix/sockets/posix_sockets.c:    if (_sockaddr_to_ep(address, address_len, &s->local) < 0) {
sys/posix/sockets/posix_sockets.c:        if (_sockaddr_to_ep(address, address_len, &r) < 0) {
sys/posix/sockets/posix_sockets.c:    if ((res = _sockaddr_to_ep(address, address_len, &ep)) < 0)
sys/posix/sockets/posix_sockets.c:                res = _sockaddr_to_ep(address, address_len, &ep);
```

So I think having it this way (for this *internal* function) is more memory efficient.

Reply:
rather use `memset(out, 0, sizeof(*out)));`?!

Reply:
Oops Oo